commit 63995653ade16deacaea5b49ceaf6376314593ac
Author: Mohammed Gamal <m.gamal005@gmail.com>
Date:   Sun Sep 19 14:34:06 2010 +0200

    KVM: Add kvm_inject_realmode_interrupt() wrapper
    
    This adds a wrapper function kvm_inject_realmode_interrupt() around the
    emulator function emulate_int_real() to allow real mode interrupt injection.
    
    [avi: initialize operand and address sizes before emulating interrupts]
    [avi: initialize rip for real mode interrupt injection]
    [avi: clear interrupt pending flag after emulating interrupt injection]
    
    Signed-off-by: Mohammed Gamal <m.gamal005@gmail.com>
    Signed-off-by: Avi Kivity <avi@redhat.com>
    Signed-off-by: Marcelo Tosatti <mtosatti@redhat.com>

diff --git a/arch/x86/kvm/x86.h b/arch/x86/kvm/x86.h
index bf4dc2f40d7f..2cea414489f3 100644
--- a/arch/x86/kvm/x86.h
+++ b/arch/x86/kvm/x86.h
@@ -72,6 +72,7 @@ static inline int is_paging(struct kvm_vcpu *vcpu)
 
 void kvm_before_handle_nmi(struct kvm_vcpu *vcpu);
 void kvm_after_handle_nmi(struct kvm_vcpu *vcpu);
+int kvm_inject_realmode_interrupt(struct kvm_vcpu *vcpu, int irq);
 
 void kvm_write_tsc(struct kvm_vcpu *vcpu, u64 data);